src/python/pants/backend/python/subsystems/setup.py

```diff
@@ -149,6 +149,20 @@ def register_options(cls, register):
                 "We recommend keeping the default of `error` for CI builds."
             ),
         )
+        register(
+            "--run-against-entire-lockfile",
+            advanced=True,
+            default=False,
+            type=bool,
+            help=(
+                "If enabled, when running binaries, tests and repls Pants will use the entire "
+                "lockfile/constraints file instead of just the relevant subset. This will improve "
+                "performance and reduce cache size, at the expense of caching granularity and "
+                "hermeticity. This option does not affect packaging deployable artifacts, such as "
```

Comment:
```suggestion
                "hermeticity.\n\nThis option does not affect packaging deployable artifacts, such as "
```

Reply:
+1 to always subsetting when packaging.

Reply:
I would add a bit more explanation here about what you mean by "hermeticity" in this context: essentially, that: unrelated thirdparty dependencies will be included in your runs/tests/repls, and that that might cause their behavior to mismatch production usage. Also, the cache key size thing.

Reply:
Hm yeah that's in particular a possible issue with runtime deps which aren't inferred. `./pants run` might work because the runtime dep is in the lockfile, but then `package` leaves it off so it fails.